File: guessing_game.kt
import kotlin.system.exitProcess

var random_number = (0..100).random()
var attempts = 0

fun generate() {
    random_number = (0..100).random()
    attempts = 0
    main()
}

fun main() {
    print("Enter a number between 1-100: ")
    var guess = readln()
    var g: Int = guess.toInt();
    attempts++
    determine(g, random_number)
}

fun determine(g: Int, random_number: Int) {
    if (g == random_number) {
        println("Correct! You got the number in $attempts attempts!")
        playagain()
    } else if (g > random_number) {
        println("Too High! Try again")
        main()
    } else if (g < random_number) {
        println("Too Low! Try again")
        main()
    }
}

fun playagain() {
    println("Would you like to play again? Y/N")
    var answer = readln()
    var stance = null
    if (answer == "Y") generate()
    else if (answer  == "N") exitProcess(0)
    else {
        println("Answer unacceptable. Please try again")
        playagain()
    }
}

File: guessing_game.go
package main
import (
	"fmt"
	"math/rand/v2"
	"strconv"
	"os"
)

var random_number int
var attempts int

func main() {
	random_number = rand.IntN(101)
	attempts = 0
	promptUser()
} 

func promptUser() {
	var guess string
	
	fmt.Println("Chose a number between 1-100")
	fmt.Scanln(&guess)
	g, err := strconv.Atoi(guess)
	if err != nil {
		fmt.Println("error")
    }
	attempts++
	identifyAccuracy(g)
}

func identifyAccuracy(g int) {
	if g == random_number {
		fmt.Println("Correct! You got the answer in " + strconv.Itoa(attempts) + " attempts")
		playAgain()
	} else if g < random_number {
		fmt.Println("Too Low, Try Again")
		promptUser()
	} else if g > random_number {
		fmt.Println("Too High, Try Again")
		promptUser()
	}
}

func playAgain() {
	var answer string
	fmt.Println("Would you like to play again? Y/N")
	fmt.Scanln(&answer)
	if answer == "Y" {
		main()
	} else if answer == "N" {
		os.Exit(1)
	} else {
		fmt.Println("Response Invalid. Try Again")
		playAgain()
	}
}